{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "944c3cda-6964-41d8-84d0-a5f38cc3d133",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%markdown\n",
    "\n",
    "## Initialize and Instantiate a File System Data Context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d05f1b1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import great_expectations as gx\n",
    "\n",
    "context = gx.get_context().convert_to_file_context()\n",
    "\n",
    "context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "64a346b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "## Connect to a Data Source (Snowflake)\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "## Connect to a Data Source (Snowflake)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31f0e317",
   "metadata": {},
   "outputs": [],
   "source": [
    "import urllib.parse\n",
    "\n",
    "# username = urllib.parse.quote(\"username\")\n",
    "# password = urllib.parse.quote(\"password\")\n",
    "# account = \"account_locator\"\n",
    "# database = \"database\"\n",
    "# schema = \"schema\"\n",
    "# warehouse = \"warehoues\"\n",
    "# role = \"role\"\n",
    "\n",
    "sf_connection_string = f\"snowflake://{username}:{password}@{account}/{database}/{schema}?warehouse={warehouse}&role={role}\"\n",
    "\n",
    "datasource_name = \"SF_DATASOURCE\"\n",
    "\n",
    "datasource = context.sources.add_or_update_snowflake(\n",
    "    name=datasource_name, \n",
    "    connection_string=sf_connection_string\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d27fb495",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "## Manage Data Assets\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "\n",
    "## Manage Data Assets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "25b8da7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SnowflakeDatasource(type='snowflake', name='SF_DATASOURCE', id=None, assets=[QueryAsset(name='customer_asset', type='query', id=None, order_by=[], batch_metadata={}, splitter=None, query='select * from customer')], connection_string=SnowflakeDsn('snowflake://hakaruop:Jenkins%40123@vr64631.central-india.azure/SNOWFLAKE_SAMPLE_DATA/TPCH_SF1?warehouse=COMPUTE_WH&role=ACCOUNTADMIN', scheme='snowflake', user='hakaruop', password='Jenkins%40123', host='vr64631.central-india.azure', tld='azure', host_type='domain', path='/SNOWFLAKE_SAMPLE_DATA/TPCH_SF1', query='warehouse=COMPUTE_WH&role=ACCOUNTADMIN'), create_temp_table=False, kwargs={})"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datasoure = context.get_datasource(datasource_name)\n",
    "\n",
    "datasource"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "41f22f0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "asset_name = \"customer_asset\"\n",
    "query = \"select * from customer\"\n",
    "\n",
    "sf_query_asset = datasource.add_query_asset(name=asset_name, query=query)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e28d7904",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "## Build Batch Request\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "\n",
    "## Build Batch Request"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ea29d37d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BatchRequest(datasource_name='SF_DATASOURCE', data_asset_name='customer_asset', options={})"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sf_batch_request = sf_query_asset.build_batch_request()\n",
    "\n",
    "sf_batch_request"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "2df07030",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'data_asset_name': 'customer_asset', 'query': 'select * from customer', 'temp_table_schema_name': None, 'batch_identifiers': {}}\n"
     ]
    }
   ],
   "source": [
    "batches = sf_query_asset.get_batch_list_from_batch_request(sf_batch_request)\n",
    "\n",
    "for batch in batches:\n",
    "    print(batch.batch_spec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "f06f9103",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "## Create Expectations\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "\n",
    "## Create Expectations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b2da83b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "suite_name = \"sf_suite\"\n",
    "suite = context.add_expectation_suite(expectation_suite_name= suite_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "b4604b27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{\"expectation_type\": \"expect_column_values_to_not_be_null\", \"kwargs\": {\"column\": \"c_name\", \"mostly\": 0.75}, \"meta\": {}}"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from great_expectations.expectations.expectation import (\n",
    "    ExpectationConfiguration,\n",
    ")\n",
    "\n",
    "c_custkey = ExpectationConfiguration(\n",
    "    expectation_type=\"expect_column_values_to_not_be_null\",\n",
    "    kwargs={\n",
    "        \"column\": \"c_custkey\",\n",
    "        \"mostly\": 0.75,\n",
    "    },\n",
    ")\n",
    "\n",
    "suite.add_expectation(\n",
    "    expectation_configuration=c_custkey\n",
    ")\n",
    "\n",
    "c_name = ExpectationConfiguration(\n",
    "    expectation_type=\"expect_column_values_to_not_be_null\",\n",
    "    kwargs={\n",
    "        \"column\": \"c_name\",\n",
    "        \"mostly\": 0.75,\n",
    "    },\n",
    "\n",
    ")\n",
    "suite.add_expectation(\n",
    "    expectation_configuration=c_name\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "b688ba82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'expectations/sf_suite.json'"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "context.save_expectation_suite(expectation_suite=suite)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "cf574dff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "## Manage a Checkpoint\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "\n",
    "## Manage a Checkpoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "9283478a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sf_checkpoint = context.add_or_update_checkpoint(\n",
    "    name=\"sf_customer_checkpoint\",\n",
    "    validations=[\n",
    "        {\n",
    "            \"batch_request\": sf_batch_request,\n",
    "            \"expectation_suite_name\": suite_name,\n",
    "        },\n",
    "    ],\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "7c49332e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6725b20091774eef88072e3e0666d851",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Calculating Metrics:   0%|          | 0/31 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "checkpoint_result = sf_checkpoint.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "aca11313",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "## playground\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%markdown\n",
    "\n",
    "## playground\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "9c94adb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'expectations/sf_suite.json'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_phone = ExpectationConfiguration(\n",
    "    expectation_type=\"expect_column_value_lengths_to_equal\",\n",
    "    kwargs={\n",
    "        \"column\": \"c_phone\",\n",
    "        \"value\":10,\n",
    "        \"mostly\": 1,\n",
    "    },\n",
    "\n",
    ")\n",
    "suite.add_expectation(\n",
    "    expectation_configuration=c_phone\n",
    ")\n",
    "\n",
    "context.save_expectation_suite(expectation_suite=suite)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "ba9da852",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error while processing DataContextConfig: expectation_suite_name\n",
      "Encountered errors during loading config.  See ValidationError for more details.\n"
     ]
    },
    {
     "ename": "InvalidDataContextConfigError",
     "evalue": "Error while processing DataContextConfig: expectation_suite_name",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mInvalidDataContextConfigError\u001b[0m             Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[69], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mgreat_expectations\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mgx\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m context \u001b[38;5;241m=\u001b[39m \u001b[43mgx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_context\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m context\u001b[38;5;241m.\u001b[39mrun_checkpoint(checkpoint_name\u001b[38;5;241m=\u001b[39msf_checkpoint)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/context_factory.py:263\u001b[0m, in \u001b[0;36mget_context\u001b[0;34m(project_config, context_root_dir, project_root_dir, runtime_environment, cloud_base_url, cloud_access_token, cloud_organization_id, cloud_mode, mode)\u001b[0m\n\u001b[1;32m    244\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mgreat_expectations\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdata_context\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdata_context\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m    245\u001b[0m     AbstractDataContext,\n\u001b[1;32m    246\u001b[0m     CloudDataContext,\n\u001b[1;32m    247\u001b[0m     EphemeralDataContext,\n\u001b[1;32m    248\u001b[0m     FileDataContext,\n\u001b[1;32m    249\u001b[0m )\n\u001b[1;32m    251\u001b[0m expected_ctx_types: \u001b[38;5;28mdict\u001b[39m[\n\u001b[1;32m    252\u001b[0m     ContextModes \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m    253\u001b[0m     Type[CloudDataContext]\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    261\u001b[0m     \u001b[38;5;28;01mNone\u001b[39;00m: AbstractDataContext,\n\u001b[1;32m    262\u001b[0m }\n\u001b[0;32m--> 263\u001b[0m context \u001b[38;5;241m=\u001b[39m \u001b[43m_get_context\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    265\u001b[0m expected_type \u001b[38;5;241m=\u001b[39m expected_ctx_types[mode]\n\u001b[1;32m    266\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(context, expected_type):\n\u001b[1;32m    267\u001b[0m     \u001b[38;5;66;03m# example I want an emphemeral context but the presence of a GX_CLOUD env var gives me a cloud context\u001b[39;00m\n\u001b[1;32m    268\u001b[0m     \u001b[38;5;66;03m# this kind of thing should not be possbile but there may be some edge cases\u001b[39;00m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/context_factory.py:302\u001b[0m, in \u001b[0;36m_get_context\u001b[0;34m(project_config, context_root_dir, project_root_dir, runtime_environment, cloud_base_url, cloud_access_token, cloud_organization_id, cloud_mode)\u001b[0m\n\u001b[1;32m    299\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m cloud_context\n\u001b[1;32m    301\u001b[0m \u001b[38;5;66;03m# Second, check for a context_root_dir to determine if using a filesystem\u001b[39;00m\n\u001b[0;32m--> 302\u001b[0m file_context \u001b[38;5;241m=\u001b[39m \u001b[43m_get_file_context\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    303\u001b[0m \u001b[43m    \u001b[49m\u001b[43mproject_config\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mproject_config\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    304\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcontext_root_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext_root_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    305\u001b[0m \u001b[43m    \u001b[49m\u001b[43mproject_root_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mproject_root_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    306\u001b[0m \u001b[43m    \u001b[49m\u001b[43mruntime_environment\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mruntime_environment\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    307\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    308\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m file_context:\n\u001b[1;32m    309\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m file_context\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/context_factory.py:383\u001b[0m, in \u001b[0;36m_get_file_context\u001b[0;34m(project_config, context_root_dir, project_root_dir, runtime_environment)\u001b[0m\n\u001b[1;32m    380\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mgreat_expectations\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdata_context\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdata_context\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m FileDataContext\n\u001b[1;32m    382\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 383\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mFileDataContext\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    384\u001b[0m \u001b[43m        \u001b[49m\u001b[43mproject_config\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mproject_config\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    385\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext_root_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext_root_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    386\u001b[0m \u001b[43m        \u001b[49m\u001b[43mproject_root_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mproject_root_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    387\u001b[0m \u001b[43m        \u001b[49m\u001b[43mruntime_environment\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mruntime_environment\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    388\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    389\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m gx_exceptions\u001b[38;5;241m.\u001b[39mConfigNotFoundError:\n\u001b[1;32m    390\u001b[0m     logger\u001b[38;5;241m.\u001b[39minfo(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCould not find local file-backed GX project\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/file_data_context.py:66\u001b[0m, in \u001b[0;36mFileDataContext.__init__\u001b[0;34m(self, project_config, context_root_dir, project_root_dir, runtime_environment)\u001b[0m\n\u001b[1;32m     60\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_context_root_directory \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_init_context_root_directory(\n\u001b[1;32m     61\u001b[0m     context_root_dir\u001b[38;5;241m=\u001b[39mcontext_root_dir,\n\u001b[1;32m     62\u001b[0m     project_root_dir\u001b[38;5;241m=\u001b[39mproject_root_dir,\n\u001b[1;32m     63\u001b[0m )\n\u001b[1;32m     64\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_scaffold_project()\n\u001b[0;32m---> 66\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_project_config \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_init_project_config\u001b[49m\u001b[43m(\u001b[49m\u001b[43mproject_config\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     67\u001b[0m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(\n\u001b[1;32m     68\u001b[0m     context_root_dir\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_context_root_directory,\n\u001b[1;32m     69\u001b[0m     runtime_environment\u001b[38;5;241m=\u001b[39mruntime_environment,\n\u001b[1;32m     70\u001b[0m )\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/file_data_context.py:111\u001b[0m, in \u001b[0;36mFileDataContext._init_project_config\u001b[0;34m(self, project_config)\u001b[0m\n\u001b[1;32m    107\u001b[0m     project_config \u001b[38;5;241m=\u001b[39m FileDataContext\u001b[38;5;241m.\u001b[39mget_or_create_data_context_config(\n\u001b[1;32m    108\u001b[0m         project_config\n\u001b[1;32m    109\u001b[0m     )\n\u001b[1;32m    110\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 111\u001b[0m     project_config \u001b[38;5;241m=\u001b[39m \u001b[43mFileDataContext\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_load_file_backed_project_config\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    112\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext_root_directory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_context_root_directory\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    113\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    114\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_apply_global_config_overrides(config\u001b[38;5;241m=\u001b[39mproject_config)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/data_context/file_data_context.py:213\u001b[0m, in \u001b[0;36mFileDataContext._load_file_backed_project_config\u001b[0;34m(cls, context_root_directory)\u001b[0m\n\u001b[1;32m    210\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m gx_exceptions\u001b[38;5;241m.\u001b[39mConfigNotFoundError()\n\u001b[1;32m    212\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 213\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mDataContextConfig\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_commented_map\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    214\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcommented_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mconfig_commented_map_from_yaml\u001b[49m\n\u001b[1;32m    215\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    216\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m gx_exceptions\u001b[38;5;241m.\u001b[39mInvalidDataContextConfigError:\n\u001b[1;32m    217\u001b[0m     \u001b[38;5;66;03m# Just to be explicit about what we intended to catch\u001b[39;00m\n\u001b[1;32m    218\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/types/base.py:139\u001b[0m, in \u001b[0;36mBaseYamlConfig.from_commented_map\u001b[0;34m(cls, commented_map)\u001b[0m\n\u001b[1;32m    137\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    138\u001b[0m     schema_instance: Schema \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m.\u001b[39m_get_schema_instance()\n\u001b[0;32m--> 139\u001b[0m     config: Union[\u001b[38;5;28mdict\u001b[39m, BYC] \u001b[38;5;241m=\u001b[39m \u001b[43mschema_instance\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcommented_map\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    140\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(config, \u001b[38;5;28mdict\u001b[39m):\n\u001b[1;32m    141\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mcls\u001b[39m\u001b[38;5;241m.\u001b[39mget_config_class()(commented_map\u001b[38;5;241m=\u001b[39mcommented_map, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mconfig)\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/marshmallow/schema.py:723\u001b[0m, in \u001b[0;36mSchema.load\u001b[0;34m(self, data, many, partial, unknown)\u001b[0m\n\u001b[1;32m    692\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mload\u001b[39m(\n\u001b[1;32m    693\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    694\u001b[0m     data: (\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    701\u001b[0m     unknown: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m    702\u001b[0m ):\n\u001b[1;32m    703\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Deserialize a data structure to an object defined by this Schema's fields.\u001b[39;00m\n\u001b[1;32m    704\u001b[0m \n\u001b[1;32m    705\u001b[0m \u001b[38;5;124;03m    :param data: The data to deserialize.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    721\u001b[0m \u001b[38;5;124;03m        if invalid data are passed.\u001b[39;00m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 723\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_do_load\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    724\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmany\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmany\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpartial\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpartial\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43munknown\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43munknown\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpostprocess\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\n\u001b[1;32m    725\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/marshmallow/schema.py:909\u001b[0m, in \u001b[0;36mSchema._do_load\u001b[0;34m(self, data, many, partial, unknown, postprocess)\u001b[0m\n\u001b[1;32m    907\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m errors:\n\u001b[1;32m    908\u001b[0m     exc \u001b[38;5;241m=\u001b[39m ValidationError(errors, data\u001b[38;5;241m=\u001b[39mdata, valid_data\u001b[38;5;241m=\u001b[39mresult)\n\u001b[0;32m--> 909\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_error\u001b[49m\u001b[43m(\u001b[49m\u001b[43mexc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmany\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmany\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpartial\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpartial\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    910\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m exc\n\u001b[1;32m    912\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m~/.python/current/lib/python3.10/site-packages/great_expectations/data_context/types/base.py:1717\u001b[0m, in \u001b[0;36mDataContextConfigSchema.handle_error\u001b[0;34m(self, exc, data, **kwargs)\u001b[0m\n\u001b[1;32m   1713\u001b[0m message: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m   1714\u001b[0m     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mError while processing DataContextConfig: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(exc\u001b[38;5;241m.\u001b[39mmessages)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1715\u001b[0m )\n\u001b[1;32m   1716\u001b[0m logger\u001b[38;5;241m.\u001b[39merror(message)\n\u001b[0;32m-> 1717\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m gx_exceptions\u001b[38;5;241m.\u001b[39mInvalidDataContextConfigError(\n\u001b[1;32m   1718\u001b[0m     message\u001b[38;5;241m=\u001b[39mmessage,\n\u001b[1;32m   1719\u001b[0m )\n",
      "\u001b[0;31mInvalidDataContextConfigError\u001b[0m: Error while processing DataContextConfig: expectation_suite_name"
     ]
    }
   ],
   "source": [
    "import great_expectations as gx\n",
    "context = gx.get_context()\n",
    "context.run_checkpoint(checkpoint_name=sf_checkpoint)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
